### PR TITLE
Rework Blankenbach and King benchmarks to compare heat flux postprocessor methods.

### PR DESCRIPTION
**Pull request by gassmoeller**
As stated in the changelog entry, this is a rework of the `blankenbach` and `king2dcompressible` benchmarks. I didnt change the setups themselves, except for adding the old way of computing heat flux values (from before #2675) for comparison and in addition to the current heat flux postprocessor (this comparison was the main purpose I started this). The new heat flux plugin for the benchmarks is exactly the old state of the heat flux before #2675 with some minor modifications I had to make to get it to work in main.

Other changes are:

1. higher resolution (up to 7 refinement now)
2. more reproducible convergence results using a python script
3. additional figures
4.  reproducible extrapolation of ASPECT reference results.

I removed the existing `Makefile`, shell scripts, and the LibreOffice table. I don't think hand written Makefiles are very readable (I found it very hard to follow), and it surprised me to see we included .ods files (even though I added this myself :-)). I also converted the README files into markdown files and included them in the documentation, so that we now have two new benchmark sections.

I am open to suggestions for improvements. I know we lost a few features with the new scripts (e.g. you cannot run individual cases without modifying the script, which was possible before), but I find it way more intuitive to read and think most of our users will prefer python scripts to Makefile syntax.

### CODE REVIEW

**Review by tjhei — APPROVED** [review]
Very nice.

File: benchmarks/blankenbach/blankenbach.png
Comment (tjhei) [inline benchmarks/blankenbach/blankenbach.png:1]: should the legend h, h^2, h^3 instead of x?

File: benchmarks/king2dcompressible/king.png
Comment (tjhei) [inline benchmarks/king2dcompressible/king.png:1]: same here

**Review by tjhei — COMMENTED** [review]
looks good otherwise

File: benchmarks/blankenbach/blankenbach.png
Comment (gassmoeller) [reply benchmarks/blankenbach/blankenbach.png:1]: I can update that. Will have to wait until tomorrow though, we shut down all computers in the lab due to the hurricane. I will let you know when its ready.

**Comment by gassmoeller** [comment]
I rebased, squashed the commits and updated the figures as requested.
